lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -267,10 +267,22 @@ module.exports = {
       return false;
     }
 
-    const oldSubscriptionFilters = response.subscriptionFilters.map((subscriptionFilter) => ({
-      ...subscriptionFilter,
-      logicalId: this.getLogicalIdFromFilterName(subscriptionFilter.filterName),
-    }));
+    const stackName = this.provider.naming.getStackName();
+    const oldSubscriptionFilters = await Promise.all(
+      response.subscriptionFilters.map(async (subscriptionFilter) => {
+        const { destinationArn, filterName } = subscriptionFilter;
+        const logicalId = this.getLogicalIdFromFilterName(filterName);
+        const isInternal = await this.provider
+          .request('CloudFormation', 'describeStackResource', {
+            StackName: stackName,
+            LogicalResourceId: logicalId,
+          })
+          .then(({ StackResourceDetail }) => filterName === StackResourceDetail.PhysicalResourceId)
```

Comment:
I know this looks concise here, but we're trying to move from promise-based approach when possible (then/catch use) and replace it with async/await if possible. Could you use `try/catch` and `await` instead of `then` and `catch` here?

Reply:
🙆‍♂️ agree with using async/await :)

+) try/catch without one-thing-function principle (like current one) makes "let" mutable variables and it makes sad.. but i will just ignore for temporaly haha

Reply:
Yeah, I know that try/catch requires to use `let` but I think in this case it's not going to be too problematic. Thank you :bow: